### PR TITLE
Allow a user to specify an initial timestamp other than 0.

### PR DESCRIPTION
**Pull request by sprousewd**
The current pyvcd implementation has an implicit initial timestamp at 0.
Some VCD traces will have signal changes beginning at times other than 0.

This  pull request allows the VCDWriter to be created with an `init_timestamp=` parameter to override the initial timestamp default of 0.

### CODE REVIEW

File: vcd/writer.py [hunk 1]
```diff
@@ -79,6 +79,7 @@ def __init__(self, file, timescale='1 us', date=None, comment='',
         self._scope_var_names = {}
         self._scope_types = {}
         self._ident_values = OrderedDict()
+        self._init_timestamp = init_timestamp
         self._prev_timestamp = 0
```
Comment (jpgrayson) [inline vcd/writer.py]: I don't think this new `_init_timestamp` instance attribute is required. Instead, I think we can just use the new `init_timestamp` parameter to initialize the existing `_prev_timestamp` attribute.

File: vcd/writer.py [hunk 2]
```diff
@@ -403,6 +404,8 @@ def _finalize_registration(self):
         assert self._registering
         print(*self._gen_header(), sep='\n', file=self._ofile)
         if self._ident_values:
+            print('#' + str(int(self._init_timestamp)), file=self._ofile)
+            self._prev_timestamp = self._init_timestamp
```
Comment (jpgrayson) [inline vcd/writer.py]: With my above suggestion to eliminate `_init_timestamp`, we can use `_prev_timestamp` for the timestamp here and eliminate this assignment.

File: vcd/writer.py (hunk 1 above)
Comment (sprousewd) [inline vcd/writer.py]: I had tried this. The problem I hit is that the signal creation/initialization technically occurs at time 0 and there are some checks that `0 < self.prev_timestamp` which will fail if we initialze `self.prev_timestamp` to something other than 0.

Comment (jpgrayson) [inline vcd/writer.py]: Could the problem have been with the `self._dump_off(0)` on line 411? That hardwired `0` is a timestamp.

Other than that, I do not see assumptions about registration being at time 0.

Even if we find that we have to keep `_init_timestamp`, that `_dump_off(0)` call needs to be addressed.

Comment (sprousewd) [inline vcd/writer.py]: Ok, let me try to address that.

Comment (sprousewd) [inline vcd/writer.py]: There's another problem. in `register_var()` there is a call to `self.change(var, 0, init)` . If we call this with `_prev_timestamp` then we will fail the `timestamp < self._prev_timestamp` check inside `change()`

Comment (sprousewd) [inline vcd/writer.py]: The problematic line is this: 
```
        if timestamp and self._registering:
            self._finalize_registration()
```
We detect the end of registration by `timestamp` being non-zero.

Comment (jpgrayson) [inline vcd/writer.py]: You are correct. The control flow in `change()` needs to be modified to accommodate. I propose:

        val_str = var.format_value(value, self._check_values)
        ts_int = int(timestamp)

        if ts_int > self._prev_timestamp:
            if self._registering:
                self._finalize_registration()
            if self._dumping:
                print('#', ts_int, sep='', file=self._ofile)
            self._prev_timestamp = ts_int

        if self._dumping and not self._registering:
            print(val_str, file=self._ofile)
        else:
            self._ident_values[var.ident] = val_str

The above passes unit tests for me.

Also, I think we should rename `_prev_timestamp` to just `_timestamp`. This better reflects the semantic of this attribute: it is the `VCDWriter`'s view of the [current] timestamp.

**Pull request merged**